### ISSUE TITLE
Iteration Plan for April 2021

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode#120585]
This plan captures our work in **April**. This is a 5-week iteration. We will ship in **early May**.

## Endgame
- **April 26, 2021**: Endgame begins
- **April 30, 2021**: Endgame done

The endgame details for this iteration are tracked [here](https://github.com/microsoft/vscode/issues?q=is%3Aissue+label%3Aendgame-plan+milestone%3A%22April+2021%22).

## Plan Items
Below is a summary of the top level plan items.

Legend of annotations:

| Mark                 | Description                                       |
| -------------------- | ------------------------------------------------- |
| :runner:             | work in progress                                  |
| :hand:               | blocked task                                      |
| :muscle:             | stretch goal for this iteration                   |
| :red_circle:         | missing issue reference                           |
| :large_blue_circle:  | more investigation required to remove uncertainty |
| :black_circle:       | under discussion within the team                  |
| :black_large_square: | a large work item, larger than one iteration      |

### Accessibility
- [x] Accessibility issues, see [query](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility+milestone%3A%22April+2021%22) @isidorn **team**
- [ ] :runner: Assess notebook accessibility https://github.com/microsoft/vscode/issues/111255 @rebornix @isidorn @roblourens

### Workbench
- [ ] :runner: Trusted Workspaces https://github.com/microsoft/vscode/issues/106488 @lszomoru @sbatten @alexdima @TylerLeonhardt @sandy081
   - [x] Finalize API
   - [x] Provide documentation for extension authors on adoption and testing
   - [x] Begin engagement with extension authors
   - [x] Trusted workspace settings https://github.com/microsoft/vscode/issues/120860 @sandy081
- [ ] :runner: Improve the welcome pages https://github.com/microsoft/vscode/issues/106715 @jacksonkearl @misolori @digitarald
   - [x] Improve UX of completed and hidden items and the recent workspaces list https://github.com/microsoft/vscode/issues/120855 @jacksonkearl @misolori
   - [ ] :runner: Improve UX for completing items https://github.com/microsoft/vscode/issues/119722 @misolori @digitarald @jacksonkearl
   - [ ] :runner: Explore extension contributions to the "New File..." item @misolori
   - [x] Allow markdown-based content in Getting Started pages https://github.com/microsoft/vscode/issues/120050, https://github.com/microsoft/vscode/issues/119709 @jacksonkearl
   - [x] Allow display of Getting Started pages on extension install @jacksonkearl
   - [ ] :muscle: Explore integration of release notes and extension changelogs @jacksonkearl
   - [ ] :runner: Continue user testing on new welcome pages @digitarald
   - [ ] :runner: Explore Extension content contributions https://github.com/microsoft/vscode/issues/118402 @digitarald
- [x] Explore UX for opening a file when there are multiple default editors registered and the user has not yet made a choice https://github.com/microsoft/vscode/issues/118063 @lramos15

### UX
- [x] Improve toolbar hover interaction/feedback https://github.com/microsoft/vscode/issues/120226 @misolori @joaomoreno
- [x] Use custom hovers in the activity bar https://github.com/microsoft/vscode/issues/118077 @sandy081
- [x] Sketch out a classification for notifications and usage guidance for each https://github.com/microsoft/vscode/issues/119463 @misolori @isidorn
- [ ] :muscle: Continue exploration of profiles https://github.com/microsoft/vscode/issues/116740 @misolori

### Notebook Editor
- [ ] :runner: Review Notebook API https://github.com/microsoft/vscode/issues/93265 @jrieken @rebornix @roblourens
- [x] Enable extensible markdown rendering for notebooks [query](https://github.com/microsoft/vscode/issues?q=is%3Aopen+label%3Anotebook-markdown+milestone%3A%22April+2021%22) @mjbvz @rebornix
- [x] Toggle line numbers of current/all cell(s) https://github.com/microsoft/vscode/issues/106054 @rebornix
- [ ] :runner: Reduce notebook papercut issues https://github.com/microsoft/vscode/issues/113245 @rebornix @jrieken @roblourens
- [x] Explore presenting moved cells in notebook diffs https://github.com/microsoft/vscode/issues/122150 @rebornix
- [ ] :runner: Adopt workspace trust for notebooks https://github.com/microsoft/vscode/issues/118584 @roblourens
- [x] Explore a declarative approach to notebook kernel and custom output renderer compatibility https://github.com/microsoft/vscode/issues/119899 @connor4312 @roblourens

### Install
- [x] `winget` support migrate to latest schema https://github.com/microsoft/vscode/issues/122272 @lszomoru

#### LSP /LSIF
- [ ] :runner: LSP - Collaborate on "TypeHierarchy" support with the Java extension team @dbaeumer **Java team**
- [ ] :runner: LSP - Add support for a pull model for diagnostics https://github.com/microsoft/language-server-protocol/issues/737 @dbaeumer
- [x] LSIF - Add support for search scopes using 'index catalogues' https://github.com/microsoft/lsif-node/issues/127 @dbaeumer

#### JavaScript/TypeScript
- [x] Prepare support for TS v4.3 @mjbvz
- [ ] :runner: Explore improving zero-knowledge onboarding into a well-tuned TS setup (tasks, tests, debugging, etc.) https://github.com/microsoft/vscode/issues/121345 @jacksonkearl @connor4312 @mjbvz @misolori

### Debug
- [x] Recommend language extensions for debugging when the user presses F5 https://github.com/microsoft/vscode/issues/119110 @isidorn
- [x] Display error response from invalid exception filter conditions https://github.com/microsoft/vscode/issues/117789 @isidorn
- [x] Add option to allow disconnect and stop/terminate UI elements https://github.com/microsoft/vscode/issues/116731 @isidorn @weinand
- [ ] :hand: Ability to drag and drop breakpoints https://github.com/microsoft/vscode/issues/88227 @isidorn
- [ ] :muscle: Investigate debug UI for highly concurrent languages https://github.com/microsoft/vscode/issues/116109 @weinand @connor4312

#### DAP
- [x] Align `setExceptionBreakpoints` with the other `setXxxException` requests https://github.com/microsoft/debug-adapter-protocol/issues/184 @weinand
- [x] Restart request should have a config argument to check if configuration changed  https://github.com/microsoft/debug-adapter-protocol/issues/180 @weinand
- [x] Provide more control over the disconnect from target behavior https://github.com/microsoft/debug-adapter-protocol/issues/175 https://github.com/microsoft/debug-adapter-protocol/issues/177 @weinand

#### JS-Debug
- [x] Explore exposing CDP commands https://github.com/microsoft/vscode-js-debug/issues/893 @connor4312
- [x] Improve discoverability of js-debug diagnostic tool https://github.com/microsoft/vscode/issues/57590 @connor4312

### Testing
- [ ] :runner: Provide consolidated testing support https://github.com/microsoft/vscode/issues/107467 @connor4312 @sandy081
- [x] Support test output https://github.com/microsoft/vscode/issues/119642 @connor4312
- [x] Support increased self-hosting with a mocha test runner https://github.com/microsoft/vscode/issues/120388 @connor4312

### Terminal
- [x] Support Terminal tabs https://github.com/microsoft/vscode/issues/10546 @tyriar @meganrogge

### Source Control
- None. See Deferred.

### API
- API finalization: [query](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+milestone%3A%22April+2021%22+label%3A%22api-finalization%22)
- API proposals: [query](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+milestone%3A%22April+2021%22+label%3A%22api-proposal%22)

### Extensions
- [x] Add option to not update disabled extensions https://github.com/microsoft/vscode/issues/76879 @sandy081

### Extension Contributions
- [x] Adopt getting started in Luna Paint https://github.com/microsoft/vscode/issues/120598 @tyriar
- [x] Improvements to the Remote extensions https://github.com/microsoft/vscode-remote-release/issues/4822 **team**
- [x] Improvements to the GitHub PR extension [query](https://github.com/microsoft/vscode-pull-request-github/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22April+2021%22) @rmacfarlane
   - [x] Improve refresh behavior of the PR views https://github.com/microsoft/vscode-pull-request-github/issues/2478 @rmacfarlane
   - [x] Explore support for resolving comments https://github.com/microsoft/vscode-pull-request-github/issues/339 @rmacfarlane
   - [x] Explore support for marking files as viewed https://github.com/microsoft/vscode-pull-request-github/issues/520 @rmacfarlane

### Engineering
- [x] Consolidate `Open with` across text editors, custom editors and notebooks, see https://github.com/microsoft/vscode/issues/100281 @lramos15
- [x] Provide file working copy handling for file based notebooks https://github.com/microsoft/vscode/issues/117873 @bpasero
- [ ] :runner: Search editor: leverage text file model support and behave like normal text editors https://github.com/microsoft/vscode/issues/118947 @jacksonkearl
- [ ] :runner: :muscle: Assume ownership of our quick picker code https://github.com/microsoft/vscode/issues/118586 @TylerLeonhardt
- [x] Explore electron 12 in Insiders https://github.com/microsoft/vscode/pull/120034 @deepak1556 @rzhao271
- [x] Explore addition of audio input, camera and apple event entitlements for integrated terminal https://github.com/microsoft/vscode/issues/119787, https://github.com/microsoft/vscode/issues/95062 @deepak1556 @tyriar
- [x] Fix `child_process.execFile` not respecting `cwd` option on macOS https://github.com/microsoft/vscode/issues/118267 @deepak1556
- [x] Support scoped Find on Page in Chromium to unblock our transition from `webview` to `iframe` for custom editors and views https://github.com/microsoft/vscode/issues/96307 @deepak1556 @rzhao271
- [x] Make `windows-process-tree` context-aware https://github.com/microsoft/vscode/issues/120570 @rzhao271
- [ ] :hand: Enable process reuse https://github.com/microsoft/vscode/issues/120431 @rzhao271
- [x] Adopt `vscode.open` and `vscode.diff` in the GHPRI extension https://github.com/microsoft/vscode-pull-request-github/issues/2333 @alexr00
- [x] Adopt `IProcessEnvService` to access shell environment https://github.com/microsoft/vscode/issues/108804 @bpasero @alexr00 @Tyriar @meganrogge
- [ ] :runner: Consolidate publishing of build artifacts https://github.com/microsoft/vscode/issues/117891 @TylerLeonhardt @joaomoreno

#### Node.js-free, sand-boxed renderer
- [ ] :runner: Backlog https://github.com/microsoft/vscode/issues/92164 @bpasero @deepak1556 @rzhao271

------
### Deferred
- [ ] Fix crash reporter to monitor only specific child processes https://github.com/microsoft/vscode/issues/118767 @deepak1556
- [ ] Scroll performance on Windows https://github.com/microsoft/vscode/issues/107016 @deepak1556
- [ ] Explore UI additions to help guide users to push to a remote after committing https://github.com/microsoft/vscode/issues/110882 @eamodio
- [ ] Improve the Process Explorer [query](https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+author%3ATyriar+label%3Aworkbench-diagnostics), https://github.com/microsoft/vscode/issues/118407 @tyriar
- [ ] Support the TS team in adopting inline value providers https://github.com/microsoft/TypeScript/issues/43449 @weinand @mjbvz
- [ ] Settings sync - debounce requests when a request is in progress https://github.com/microsoft/vscode/issues/91286 @sandy081
